components/micro-gateway-core/src/main/ballerina/gateway/filters/subscription_filter.bal

```diff
@@ -37,134 +37,131 @@ public type SubscriptionFilter object {
-                            return true;
-                        }
-                        foreach var subscription in subscribedAPIList {
-                            if (subscription.name.toString() == apiConfig.name &&
-                                subscription["version"].toString() == apiConfig.apiVersion) {
-                                printDebug(KEY_SUBSCRIPTION_FILTER, "Found a matching subscription with name:" +
-                                        subscription.name.toString() + " version:" + subscription["version"].
-                                        toString());
-                                authenticationContext.authenticated = true;
-                                authenticationContext.tier = subscription.subscriptionTier.toString();
-                                authenticationContext.apiKey = jwtToken;
-                                authenticationContext.username = decodedPayload.sub.toString();
-                                authenticationContext.callerToken = jwtToken;
-                                authenticationContext.applicationId = decodedPayload.application.id.toString();
-                                authenticationContext.applicationName = decodedPayload.application.name.toString
-                                ();
-                                authenticationContext.applicationTier = decodedPayload.application.tier.toString
-                                ();
-                                authenticationContext.subscriber = decodedPayload.application.owner.toString();
-                                authenticationContext.consumerKey = decodedPayload.consumerKey.toString();
-                                authenticationContext.apiTier = subscription.subscriptionTier.toString();
-                                authenticationContext.apiPublisher = subscription.publisher.toString();
-                                authenticationContext.subscriberTenantDomain = subscription
-                                .subscriberTenantDomain.toString();
-                                authenticationContext.keyType = decodedPayload.keytype.toString();
-                                // setting keytype to invocationContext
-                                printDebug(KEY_SUBSCRIPTION_FILTER, "Setting key type as " +
-                                        authenticationContext.keyType);
-                                runtime:getInvocationContext().attributes[KEY_TYPE_ATTR] = authenticationContext
-                                .keyType;
-                                filterContext.attributes[AUTHENTICATION_CONTEXT] = authenticationContext;
-                                printDebug(KEY_SUBSCRIPTION_FILTER, "Subscription validation success.");
-                                return true;
-                            }
-                        }
+        if (authScheme != AUTH_SCHEME_JWT){
```

Comment:
This will continue to the logic for auth schemes which are not jwt. We need to skip this filter when auth scheme is not jwt

Reply:
Good catch! Fixed it.